File: abstractclass.java
public class abstractclass {
    public static void main(String[] args) {
        Rectangle rectangle = new Rectangle(4, 6);
        System.out.println("Area of Rectangle: " + rectangle.calculateArea());
    }
}
abstract class Shape {
    public void display() {
        System.out.println("This is a shape.");
    }
    abstract public double calculateArea();
}
class Rectangle extends Shape {
     int  length, width;

    Rectangle(int length, int width) {
        this.length = length;
        this.width = width;
    }
    @Override
    public double calculateArea() {
        return length * width;
    }
}
 

File: dart/abstract_class.dart
//Abstract class
import 'dart:math' show pi;

abstract class Shape{
  double area();
  void display(){
    print("Concrete method from abstract class");
  }
}

class Circle extends Shape{
  dynamic radius;
  Circle({this.radius});
  double area(){
    var a=pi*radius*radius;
    return a;
  }
}

class Square extends Shape{
  dynamic length;
  Square({this.length});
  double area(){
    var b=length*length;
    return b;
  }
}

void main(){
  Circle cobj=Circle(radius:30.0);
  print('area of circle: ${cobj.area()}');
  Square sobj=Square(length:10.0);
  print('area of square: ${sobj.area()}');
  cobj.display();
}